app/src/main/java/framgia/com/mynote/screen/detail_note/DetailAdapter.java

```diff
@@ -0,0 +1,80 @@
+package framgia.com.mynote.screen.detail_note;
+
+
+import android.content.Context;
+import android.databinding.DataBindingUtil;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+
+import java.util.List;
+
+import framgia.com.mynote.R;
+import framgia.com.mynote.data.model.Task;
+import framgia.com.mynote.databinding.ItemTaskBinding;
+
+public class DetailAdapter extends RecyclerView.Adapter<DetailAdapter.ViewHolder>
+        implements TaskItemUserActionListener {
+    private List<Task> mTasks;
+    private DetailViewModel mViewModel;
+    private LayoutInflater mInflater;
+
+    public DetailAdapter(Context context, List<Task> tasks, DetailViewModel viewModel) {
+        mViewModel = viewModel;
+        mInflater = LayoutInflater.from(context);
+        mTasks = tasks;
+    }
+
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
+        ItemTaskBinding binding = DataBindingUtil
+                .inflate(mInflater, R.layout.item_task, viewGroup, false);
+        binding.textTaskTitle.setFocusable(false);
+        return new ViewHolder(binding);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull DetailAdapter.ViewHolder viewHolder, int i) {
+        viewHolder.mBinding.setTask(mTasks.get(i));
+        viewHolder.mBinding.setListener(this);
+    }
+
+    @Override
+    public int getItemCount() {
+        return mTasks == null ? 0 : mTasks.size();
+    }
+
+    @Override
+    public void onCheckedChanged(Task task, boolean isChecked) {
+        task.setDone(isChecked ? 1 : 0);
+        mViewModel.getCheckBoxTaskEvent().setValue(task);
```

Comment:
sao lại cần gọi cả 2 event cùng 1 lúc em nhỉ

Reply:
```suggestion
        mViewModel.getCheckBoxTaskEvent().setValue(task);
```
đầu tiên em sẽ phải gán lại giá trị cho task  thành done hoặc not done 
sau đó em sẽ phải update lại dữ liệu của task ở trong sql ạ